### PR TITLE
Fixed writeable name from range to geo_distance

### PR DESCRIPTION
**Pull request by brandonkearby**
Fixed writeable name from range to geo_distance

### CODE REVIEW

**Comment by s1monw** [comment]
wow how did you figure out about this? any chance you can provide a test case. I also see you didn't sign the CLA can you do so?

**Comment by brandonkearby** [comment]
Hi, @s1monw I'm currently working on a high-level rest client and was testing it against the master branch. That's how I figured it out. My unit test failed. [https://github.com/brandonkearby/elasticsearch/blob/rest-client-1.4.1/src/test/java/org/elasticsearch/client/rest/RestClientTest.java](RestClientTest)

I just signed the CLA. It's under my work email: bkearby@sprinklr.com.

**Comment by nik9000** [comment]
This looks right to me and I've tentatively labeled this as an aggregations bug. Can you add a unit or integration test or *something* that fails without the fix? `org.elasticsearch.search.aggregations.bucket.GeoDistanceIT` is an integration test with assertions about the geo distance aggregation. I'm happy to help in any way you'd like. I expect I was the one who broke this in the first place.

**Comment by nik9000** [comment]
I'll have a look at the cla thing.

**Comment by nik9000** [comment]
Poking the ticket so the CLA checker will recheck.

**Comment by karmi** [comment]
Hi @brandonkearby, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch/pull/20860.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by brandonkearby** [comment]
Hi @karmi  Just added my work email to my github account as a secondary email.

**Comment by brandonkearby** [comment]
Added my git commit email as well.

**Comment by nik9000** [comment]
@brandonkearby that did the trick!

**Comment by brandonkearby** [comment]
WooT

**Comment by brandonkearby** [comment]
Hi @nik9000, I'll work on adding a test that will fail.

**Comment by nik9000** [comment]
Thanks!

File: core/src/test/java/org/elasticsearch/search/geo/GeoDistanceIT.java [hunk 1]
```diff
@@ -156,4 +164,33 @@ public void testDistanceScript() throws Exception {
...
+            .unit(DistanceUnit.MILES)
+            .addRange(0, 2000));
+
+        search.setSize(0); // no hits please
+
+        SearchResponse response = search.get();
+        Aggregations aggregations = response.getAggregations();
+        assertNotNull(aggregations);
+        InternalGeoDistance geoDistance = aggregations.get(name);
+        assertNotNull(geoDistance);
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/geo/GeoDistanceIT.java]: Could you assert the result is correct while you are here?

**Review by nik9000 — CHANGES_REQUESTED** [review]

**Comment by nik9000** [comment]
@brandonkearby thanks for adding the test! I've reviewed it and left a comment - just extra coverage on top of the "did it blow up" test that you added to catch this error. If it is too much trouble I'm happy to do it myself in a followup but if you can do it it'd be lovely!

**Comment by brandonkearby** [comment]
Hi @nik9000, I've added a test. One interesting thing I noticed is InternalGeoDistance use to implement GeoDistance (interface). Now it no longer implements it as it's been converted to an enum. In the test provided I'm casting to InternalGeoDistance just to prove that class isn't the one we are getting.

**Comment by nik9000** [comment]
> One interesting thing I noticed is InternalGeoDistance use to implement GeoDistance (interface). Now it no longer implements it as it's been converted to an enum.

I'm not really sure what happened there. I'm not super sure what the vision of the transport API was for these classes. For now the cast is fine.

**Comment by brandonkearby** [comment]
Hi @nik9000, Added asserts, but they are failing :(

**Comment by nik9000** [comment]
Fun! I'll fetch it locally in a few minutes and have a look. Thanks so much!

**Comment by nik9000** [comment]
@brandonkearby I think the second test needs to set up the mapping like this: https://gist.github.com/nik9000/29d0b18f5dde28d3e76b26d1d28f67f1

**Comment by brandonkearby** [comment]
@nik9000 Yep, that fixed it! 👍

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
OK! Looks good to me! I'll merge and backport it in a few minutes.

**Comment by brandonkearby** [comment]
@nik9000 Perfect thanks!

**Comment by nik9000** [comment]
@brandonkearby the build fails for me because of the wildcard import - we disallow those in the code base because they make it more difficult for our non-IDE using friends to track to track down imports and it is fairly easy for those of us that do use IDEs to teach them not to add them. I'll remove the wildcard import before merging and backporting.

**Comment by nik9000** [comment]
> I'll remove the wildcard import before merging and backporting.

Or you can if you like, either way.

**Comment by brandonkearby** [comment]
@nik9000 Thanks I'll change my ide settings for next time :)

**Comment by nik9000** [comment]
> @nik9000 Thanks I'll change my ide settings for next time :)

Awesome.

**Comment by nik9000** [comment]
OK! I've merged into master and I've staged the backports to 5.x and am running tests locally for that. I'll edit this comment with the hashes once I've pushed that and do the same for 5.0. Thanks for fixing this @brandonkearby!


master: a3908c4c406e29e38b44dd74a112ba3564898202 (+ cb8fe00b0ebd16a0bd12106ed6785eabac2b658b)
5.x:
5.0:

**Comment by nik9000** [comment]
And done! Thanks again @brandonkearby !

**Comment by brandonkearby** [comment]
Anytime @nik9000 glad to help :)

**Pull request merged**